xbmc/cores/AudioEngine/Sinks/AESinkDARWINOSX.cpp

```diff
@@ -93,13 +93,56 @@ static CADeviceList GetDevices()
   return list;
 }
 
+//---------- DEVICE CHANGED EVENT HANDLING
+// when device list change callbacks are received
+// inform the ae engine this amount of time after
+// the last callback was received (bundle multiple callbacks)
+// into one final call
+#define DEVICE_LIST_CHANGED_TIMER_MS 1500
+
+CDeviceChangeListTimeout::CDeviceChangeListTimeout()
+{
+  m_pDeviceListChangedTimer = new CTimer(this);
```

Comment:
Also that timer can't be a member of the sink class because the sink object can be deleted by AE due to the fact that the device that sink has currently open was removed (i had this approach checked and wondered why my timer didn't fire - it was because AE timedout while adding packets to the sink and destroyed the sink object - which also destroyed the timer which was supposed to fire and notify AE of the changed device list).